### DISCUSSION TITLE
non-Wayland alternative

### DISCUSSION THREAD

**Discussion by jelabarre59** [waydroid/waydroid#786] [Q&A]
Since Waydroid is designed for Wayland environments, can anyone suggest an equivalent application to work on X11?  Cinnamon is a long ways from having Wayland support (and no, I'm not changing desktops).

**Comment by YogSottot** [comment]
You can just use weston to run waydroid in x11.

**Comment by TriMoon** [comment]
> You can just use weston to run waydroid in x11.

Do mean that Weston can act like a Wayland server under X11?

Would be handy for reference if you gave a step-by-step example of what is needed and how to actually do that.
_(Disclaimer: This was first time I learned about [Weston](https://wiki.archlinux.org/title/Weston))_

**Reply by YogSottot** [reply]
```
#!/bin/bash
set -xeuo pipefail
trap "kill 0" SIGINT;

killall -9 weston || true;
sudo systemctl restart waydroid-container.service;

if ! pgrep weston; then
     weston &> /dev/null &
     PID1=$!
fi
sleep 2;

waydroid show-full-ui &

wait $PID1
waydroid session stop ;
sudo systemctl stop waydroid-container.service ;
killall -9 weston || true;
```

**Reply by harlandski** [reply]
How should you use this script? If I run weston, then run the script, then weston gets closed. If I run it without weston running then I get the usual Wayland socket '/run/user/1000/wayland-0' doesn't exist; are you running a Wayland compositor? error

**Reply by YogSottot** [reply]
I updated the script.
When copying my script, I deleted optional things and accidentally deleted ```wait $PID1```

**Reply by harlandski** [reply]
Hey thanks for replying. I'm getting PID1: unbound variable now.

I wonder if something else is missing from your fuller version?

**Reply by YogSottot** [reply]
I copied this script and ran it. It works. Copy it in its entirety. There is ```PID1=$!``` in ```if``` block

**Reply by harlandski** [reply]
Yeah, I've copied it all, and also entered everything line-by line doing the logic myself, but in both cases I'm getting OSError:container failed to start. Maybe I've not installed weston properly.

Edit: I installed some dev libraries for Weston, still no joy.

Here are the error messages I'm getting:

[21:29:25] Starting waydroid session
[21:29:25] WAYLAND_DISPLAY is not set, defaulting to "wayland-0"
[21:29:36] OSError: container failed to start

I wonder if I need to set WAYLAND_DISPLAY to interact with Weston

**Reply by YogSottot** [reply]
>[21:29:36] OSError: container failed to start

This is a waydroid problem, not a weston problem. See the waydroid log.
https://docs.waydro.id/debugging/getting-essential-information

> I wonder if I need to set WAYLAND_DISPLAY to interact with Weston

Not necessarily, "wayland-0" is enough.

**Reply by harlandski** [reply]
OK, cool, I'll see what is wrong there then, thanks! By the way by using weston --socket=wayland-0 I eliminated the first error message.

**Reply by StephanOpfer** [reply]
Hi There,
I followed your conversation and and tried it myself. Unfortunately, I cannot report any success. 
@harlandski is it working for you?

**Reply by gitmanbloggus** [reply]
> Hi There,
> I followed your conversation and and tried it myself. Unfortunately, I cannot report any success. 
> @harlandski is it working for you?

I manually started weston then ran "waydroid show-full-ui" in weston
works for me
